### PR TITLE
update trafficcontroller decorator to handle container metrics split …

### PR DESCRIPTION
**Pull request by kitemongerer**
…across several envelopes

Signed-off-by: Tom Chen <tochen@pivotal.io>

Thanks for contributing to cloud_controller_ng. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:
We updated the trafficcontroller decorator to be able to handle both the case where all container metrics are in one envelope and the case where container metrics are split across envelopes

* An explanation of the use cases your change solves
Log cache will be switching to ingressing via syslog which will cause the container metrics to be in separate envelopes

* Links to any other associated PRs

* [x] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/CONTRIBUTING.md)

* [x] I have viewed, signed, and submitted the Contributor License Agreement

* [x] I have made this pull request to the `master` branch

* [x] I have run all the unit tests using `bundle exec rake`

* [x] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats)

### CODE REVIEW

File: lib/logcache/traffic_controller_decorator.rb [hunk 1]
```diff
@@ -60,23 +62,43 @@ def has_container_metrics_fields?(envelope)
       # rubocop seems to think that there is a 'key?' method
       # on envelope.gauge.metrics - but it does not
       # rubocop:disable Style/PreferredHashMethods
-      envelope.gauge.metrics.has_key?('cpu') &&
-      envelope.gauge.metrics.has_key?('memory') &&
+      envelope.gauge.metrics.has_key?('cpu') ||
```
Comment (selzoc) [inline lib/logcache/traffic_controller_decorator.rb]: Why was this changed from `&&` to `||`?

File: spec/logcache/traffic_controller_decorator_spec.rb [hunk 2]
```diff
@@ -399,6 +399,60 @@ def generate_batch(size, offset: 0, last_timestamp: TimeUtils.to_nanoseconds(Tim
...
+
+        expect(subject.first.containerMetric.instanceIndex).to eq(1)
+        expect(subject.first.containerMetric.cpuPercentage).to eq(10)
+        expect(subject.first.containerMetric.memoryBytes).to eq(21)
+        expect(subject.first.containerMetric.diskBytes).to eq(32)
+
+        expect(subject.second.containerMetric.instanceIndex).to eq(2)
+        expect(subject.second.containerMetric.cpuPercentage).to eq(30)
+        expect(subject.second.containerMetric.memoryBytes).to eq(31)
+        expect(subject.second.containerMetric.diskBytes).to eq(32)
```
Comment (selzoc) [inline spec/logcache/traffic_controller_decorator_spec.rb]: This should probably be a distinct value from the first instance's `diskBytes`

File: lib/logcache/traffic_controller_decorator.rb (hunk 1 above)
Comment (selzoc) [reply lib/logcache/traffic_controller_decorator.rb]: duh, i read further

File: spec/logcache/traffic_controller_decorator_spec.rb [hunk 3]
```diff
@@ -399,6 +399,60 @@ def generate_batch(size, offset: 0, last_timestamp: TimeUtils.to_nanoseconds(Tim
...
+                  'disk' => Loggregator::V2::GaugeValue.new(unit: 'bytes', value: 32),
+              }),
+              instance_id: '1'
+            ),
+            Loggregator::V2::Envelope.new(
+              source_id: process_guid,
+              gauge: Loggregator::V2::Gauge.new(metrics: {
+                  'cpu' => Loggregator::V2::GaugeValue.new(unit: 'bytes', value: 30),
+                  'memory' => Loggregator::V2::GaugeValue.new(unit: 'bytes', value: 31),
+                  'disk' => Loggregator::V2::GaugeValue.new(unit: 'bytes', value: 32),
```
Comment (selzoc) [inline spec/logcache/traffic_controller_decorator_spec.rb]:
```suggestion
                  'disk' => Loggregator::V2::GaugeValue.new(unit: 'bytes', value: 34),
```

File: spec/logcache/traffic_controller_decorator_spec.rb (hunk 2 above)
Comment (selzoc) [inline spec/logcache/traffic_controller_decorator_spec.rb]:
```suggestion
        expect(subject.second.containerMetric.diskBytes).to eq(34)
```

**Comment by kitemongerer** [comment]
@selzoc We made the changes you suggested. Thank you!
